### PR TITLE
Tuning Passenger configuration file

### PR DESCRIPTION
**Pull request by Aethylred**
This patch modifies `apache::mod::passenger` so that it creates a `passenger.conf` in the modules directory using a template. These are global parameters for the passenger instance, hence should be attached to the module and not to a vhost configuration.

Only a subset of the parameters have been created, these are sufficient for setting up a puppetmaster with apache::mod::passenger.

### CODE REVIEW

**Comment by Aethylred** [comment]
excellent, and I've correctly identified a common node to rebase onto dc11bfdfd32020a7ede693c280a69df601c476ee :)

**Comment by hunner** [comment]
Thanks @Aethylred! I'll get to merging this as soon as we have the current branch stabilized.

Note for myself when I get to merging this: rebase onto released refactored branch and add parameter descriptions to Lauren's updated docs.

File: manifests/init.pp [hunk 1]
```diff
@@ -65,8 +65,10 @@
       ensure  => directory,
       require => Package['httpd'],
     } -> A2mod <| |>
-    resources { 'a2mod':
-      purge => true,
+    if $::osfamily != 'debian' {
```
Comment (Aethylred) [inline manifests/init.pp]: This block may no longer be required.

**Comment by Aethylred** [comment]
ok.... this looks a bit messy, I've merged it with the new master from puppetlabs/puppetlabs-apache 

...and added some spec tests.

If you update refactor_module it should clean up.

**Comment by Aethylred** [comment]
That should be the complete testing that whatever you pass to the parameter ends up in the passenger.conf file.

**Pull request closed without merging**